idea-plugin/src/main/kotlin/arrow/meta/ide/phases/resolve/MetaSyntheticPackageFragmentProvider.kt

```diff
@@ -135,340 +62,39 @@ class MetaSyntheticPackageFragmentProvider(val project: Project) :
 
     inner class Scope : MemberScope {
       override fun getClassifierNames(): Set<Name>? =
-        descriptorCache[fqName]?.filterIsInstance<ClassifierDescriptor>()?.map { it.name }?.toSet()
+        cache.resolved(fqName)?.filterIsInstance<ClassifierDescriptor>()?.map { it.name }?.toSet()
 
       override fun getContributedClassifier(name: Name, location: LookupLocation): ClassifierDescriptor? =
-        descriptorCache[fqName]?.filterIsInstance<ClassifierDescriptor>()?.firstOrNull { it.name == name }
+        cache.resolved(fqName)?.filterIsInstance<ClassifierDescriptor>()?.firstOrNull { it.name == name }
 
       override fun getContributedDescriptors(kindFilter: DescriptorKindFilter, nameFilter: (Name) -> Boolean): Collection<DeclarationDescriptor> =
-        descriptorCache[fqName]?.filter { it.name == name } ?: emptyList()
+        cache.resolved(fqName)?.filter { it.name == name } ?: emptyList()
```

Comment:
Not that it really matters but if you wanted to get rid of these Elvis operators you could potentially just chain orEmpty() which should be available in all nullable types collections.